src/components/form/Calendar/index.tsx

```diff
@@ -0,0 +1,131 @@
+import { Dispatch, SetStateAction, useEffect, useRef } from 'react';
+import { styled } from '@stitches/react';
+import Calendar from 'react-calendar';
+import 'react-calendar/dist/Calendar.css';
+import dayjs from 'dayjs';
+import ErrorMessage from '../ErrorMessage';
+import { useDisplay } from '@hooks/useDisplay';
+import BottomSheetDialog from '../Select/BottomSheetSelect/BottomSheetDialog';
+interface Props {
+  selectedDate: string | null;
+  setSelectedDate: Dispatch<SetStateAction<string | null>>;
+  isOpen: boolean;
+  setIsOpen: Dispatch<SetStateAction<boolean>>;
+  error?: string;
+}
+
+const CalendarModal = ({ selectedDate, setSelectedDate, isOpen, setIsOpen, error }: Props) => {
+  const containerRef = useRef<HTMLDivElement | null>(null);
+  const { isDesktop, isMobile, isTablet } = useDisplay();
+
+  const handleOutsideClick = (event: any) => {
+    if (!containerRef.current || !containerRef.current.contains(event.target)) {
+      setIsOpen(false);
+    }
+  };
+
+  useEffect(() => {
+    if (isDesktop && !isMobile && !isTablet) {
+      document.addEventListener('mousedown', handleOutsideClick);
+    }
+    return () => {
+      if (isDesktop && !isMobile && !isTablet) {
+        document.removeEventListener('mousedown', handleOutsideClick);
+      }
+    };
```

Comment:
```suggestion
    if (isDesktop && !isMobile && !isTablet) {
      document.addEventListener('mousedown', handleOutsideClick);
      
      return () => {
        document.removeEventListener('mousedown', handleOutsideClick);
      };
    }
```
useEffect 의 의존성 배열에 isMobile 과 isTablet, handleOutsideClick 을 넣어줄 필요가 있어보여요!! 이유는 아래 문서를 참조해주세요 :)
https://ko.react.dev/reference/react/useEffect#specifying-reactive-dependencies
또, 같은 조건문에서 return 시키는 것이라면 위와 같이 코드를 수정하면 더 깔끔해보일 것 같아요!!

Reply:
지당한 말씀입니다. 참고해서 수정하도록 할게요!